query/expression.go

```diff
@@ -116,6 +118,8 @@ func (expr stringExpression) Evaluate(context EvaluationContext) (value, error)
 }
 
 func (expr *metricFetchExpression) Evaluate(context EvaluationContext) (value, error) {
+	// Record the duration of this request in the profiler.
+	defer context.Profiler.Add("metricFetchExpression.Evaluate()", inspect.Now())
```

Comment:
from http://blog.golang.org/defer-panic-and-recover

"1. A deferred function's arguments are evaluated when the defer statement is evaluated."

The function itself is called at the end, but the `inspect.Now()` is called here.